gb/src/terminal_debugger.rs

```diff
@@ -0,0 +1,118 @@
+use std::{sync::atomic::{AtomicBool, Ordering}, io::stdin, thread};
+
+use crossbeam_channel::{bounded, Sender, Receiver};
+use lib_gb::{machine::debugger::{DebuggerUi, DebuggerCommand, DebuggerResult}};
+
+static ENABLE_FLAG: AtomicBool = AtomicBool::new(false);
```

Comment:
This makes the debugger a singleton, either remove this or protect it against cases it spawns more than 1 instance